src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -494,15 +491,43 @@ private URI generateAcceptRequestUri(URI rendezvousUri) throws URISyntaxExceptio
 			RelayLogger.logEvent("rendezvousStop", this);
 		});
 	}
+	
+	void clearFault() {
+        this.injectedFault = null;
+	}
+	
+	/**
+	 * Inject a specific exception to the listener or its underlying websocket for testing purposes
+	 * @param throwable The following exceptions can be interpreted:
+	 * </br><b>UpgradeException</b>    - When this exception is injected, all listener websocket connection attempts will fail 
+	 *                                   until the fault has been cleared.
+	 * </br><b>ConnectionException</b> - Injecting this exception will kill the listener's websocket (simulate a network disconnect),
+	 *                                   but should cause the listener to try to reconnect.
+	 */
```

Comment:
Cool stuff!

Reply:
If there are more faults needed in the future we just add here. I wonder if there are better ways to doing this though.

---
In reply to: [354520748](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354520748) [](ancestors = 354520748)

Reply:
In the future maybe have an enum specifying when to use the exception plus an exception. That would allow throwing the same exception type from different locations, for example.

---
In reply to: [354522602](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354522602) [](ancestors = 354522602,354520748)